### DISCUSSION TITLE
error building documentation

### DISCUSSION THREAD

**Discussion by nlosacco** [idaholab/moose#23257] [Q&A Tools] [answered]
Hi all,

I'm trying to build the html documentation for MOOSE on my computer (Mac OSX 12.4 Monterey). That would be very useful cause I'm working offline quite often.

When I do it from the "test" directory:

`cd /home/user/projects/moose/test
./moosedocs.py build --destination [destination_dir]`

it works smoothly. But as far as I understand, in order to replicate the documentation of the actual MOOSE website I must run the command in the "modules" directory.
When I do, I get:

`MooseDocs.extensions.appsyntax (MainProcess): Failed to locate a valid executable in ${MOOSE_DIR}/modules/combined.`

and then a lot of other errors about syntax not being found.

What am I doing wrong?

Thanks for any help,

N

**Comment by hugary1995** [comment, marked as answer]
The script looks for combined-opt and couldn't find it. You need to compile combined-opt in modules/combined.

**Reply by milljm** [reply]
You need to build moose/modules, and then run your moosedocs command from there. Like so:

```bash
cd moose/modules
make -j 6
cd doc
./moosedocs.py build --destination <destination dir>
```

**Reply by nlosacco** [reply]
that's great, thank you.
Now I wonder if it is possible to embed a local search engine, rather than the google driven one, which tries and connect to the web
